modules/wpml/wpml-config.php

```diff
@@ -582,32 +598,42 @@ private function get_field_attribute( SimpleXMLElement $field, $attribute_name )
 	 *
 	 * @param  SimpleXMLElement $field A XML node.
 	 * @param  array            $attrs The attributes.
-	 * @return array|string An array if there are sub-attributes, otherwise a string.
+	 * @return array An array of attributes.
+	 *
+	 * @phpstan-param array<non-empty-string, array|true> $attrs
+	 * @phpstan-return array<non-empty-string, array|true>
 	 */
-	private function get_field_attributes( SimpleXMLElement $field, array $attrs = array() ) {
+	private function get_field_attributes( SimpleXMLElement $field, array $attrs = array() ): array {
 		$name = $this->get_field_attribute( $field, 'name' );
 
+		if ( '' === $name ) {
+			return array();
+		}
+
 		$children = $field->children();
 
-		if ( 0 === count( $children ) ) {
-			// No sub-attributes, don't go further and just return the attribute's value.
-			return $name;
+		if ( 0 === $children->count() ) {
+			return array( $name => true );
 		}
 
-		foreach ( $children as $child ) {
-			if ( ! isset( $attrs[ $name ] ) || ! is_array( $attrs[ $name ] ) ) {
-				$attrs[ $name ] = array();
-			}
+		if ( ! isset( $attrs[ $name ] ) || ! is_array( $attrs[ $name ] ) ) {
+			$attrs[ $name ] = array();
+		}
```

Comment:
```suggestion
		$sub_attributes = array();
		if ( isset( $attrs[ $name ] ) && is_array( $attrs[ $name ] ) ) {
			$sub_attributes = $attrs[ $name ];
		}
```
So after there is no need to unset `$attrs[ $name ]` at the end.

Reply:
You could check `isset( $attrs[ $name ] )` and then do the `array_merge()` or create it otherwise.
But I let you see what you prefer.

Reply:
I'll keep it as it is, what you suggest complicates things actually because we have to do the `if ( isset() && is_array() ) {` twice.

Reply:
@Hug0-Drelon I think I find a way to simplify it with [73298ee](https://github.com/polylang/polylang/pull/1434/commits/73298eebb39035a47d1a3a1d4afe334830f136d2).